src/controllers/dlgprefcontroller.cpp

```diff
@@ -491,30 +496,133 @@ void DlgPrefController::savePreset() {
+            saveAsNew = false;
+            if (alwaysOverwrite.checkState() == Qt::Checked) {
+                m_pOverwritePresets.insert(m_pPreset->filePath(), true);
+            }
+        } else if (overwriteDialog.close()) {
+            return;
+        }
+    }
 
-    // Add " (edited)" to preset name (if it's not already present)
-    QString editedSuffix = QStringLiteral(" (") + tr("edited") + QStringLiteral(")");
-    if (!m_pPreset->name().endsWith(editedSuffix)) {
-        m_pPreset->setName(m_pPreset->name() + editedSuffix);
-        qDebug() << "Renamed preset to " << m_pPreset->name();
+    // Initially save a modified Mixxx preset to the user folder or save a user mapping
+    // with a new name.
+    if (saveAsNew) {
+        QString savePresetTitle = QObject::tr("Save user preset");
+        QString fileNameLabel = QObject::tr("Enter the file name for the preset:");
+        QString displayNameLabel = QObject::tr("Enter the display name for the preset:");
+
+        QString savingFailedTitle = QObject::tr("Saving preset failed");
+        QString renamingFailedTitle = QObject::tr("Renaming Preset Failed");
+        QString fileExistsLabel = QObject::tr("A preset with that name already exists.");
+        QString savingFailedLabel = QObject::tr(
+                "A preset cannot have a blank name and may not contain special characters.");
+
+        bool validBaseName = false;
+        bool validDisplayName = false;
+
+        // Choose a new file (base) name
+        while (!validBaseName) {
+            QString userDir = m_pUserDir;
+            bool ok = false;
+            baseName = QInputDialog::getText(nullptr,
+                    savePresetTitle,
+                    fileNameLabel,
+                    QLineEdit::Normal,
+                    baseName,
+                    &ok)
+                               .trimmed()
+                               .remove(rx);
+            if (!ok) {
+                return;
+            }
+            if (baseName.isEmpty()) {
+                QMessageBox::warning(nullptr,
+                        savingFailedTitle,
+                        savingFailedLabel);
+                continue;
+            }
+            filePath = userDir + baseName + kPresetExt;
+            if (QFile::exists(filePath)) {
+                QMessageBox::warning(nullptr,
+                        savingFailedTitle,
+                        fileExistsLabel);
+                continue;
+            }
+            validBaseName = true;
+        }
 
-        // Add " (edited)" to file name (if it's not already present)
-        QString baseName = fileInfo.baseName();
-        if (baseName.endsWith(editedSuffix)) {
-            baseName.chop(editedSuffix.size());
+        // Choose a new name for the preset selector and the preset description
+        while (!validDisplayName) {
+            bool ok = false;
+            displayName = QInputDialog::getText(nullptr,
```

Comment:
Could you merge filename and display name into a single dialog? At first I suspected a bug because it asks me twice...

Reply:
Done. Forgot to merge those two after dismissing the QFileDialog.

Now the dialog picks the preset name as default text for the lineedit and (besides removing invalid characters) reformats the entered name so it be used as file path (replace `/` `\` with `-`).